pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -36,39 +37,61 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
 	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 )
 
+const (
+	eventChBufferSize = 10
+
+	// this set will provide waiting time between 5s to 5min
+	retryIncrement = 5 * time.Second
+	retryMaxSteps  = 6
+)
+
+// retryAfter returns an exponentially increasing interval between
+// retryIncrement and 2^retryMaxSteps * retryIncrement
+func retryAfter(failedAttempts uint) time.Duration {
+	return (1 << min(failedAttempts, retryMaxSteps)) * retryIncrement
```

Comment:
the attempts are incremented after this is called , we start with 0 not one

Reply:
I see, I think it would be more readable if we already incremented the failure counter before calling this function. It is called in reaction to failure, so feel weird calling it with failure counter=0.

Reply:
done